KeyVault.Acmebot/Providers/NamecheapProvider.cs

```diff
@@ -0,0 +1,561 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using System.Xml;
+
+using KeyVault.Acmebot.Internal;
+using KeyVault.Acmebot.Options;
+
+namespace KeyVault.Acmebot.Providers;
+
+public class NamecheapProvider : IDnsProvider
+{
+    public const string NamecheapXmlNamespace = @"http://api.namecheap.com/xml.response";
+    public const string Endpoint = @"https://api.namecheap.com/xml.response";
+    public const string IpfyOrgEndpoint = @"https://api.ipfy.org";
+    public const string ListDomainsCommand = "namecheap.domains.getList";
+    public const string GetHostsCommand = "namecheap.domains.dns.getHosts";
+    public const string SetHostsCommand = "namecheap.domains.dns.setHosts";
+
+    public NamecheapProvider(NamecheapOptions options, HttpClient namecheapHttpClient, HttpClient ipfyOrgHttpClient)
```

Comment:
As with other Providers, the implementation of the IDnsProvider interface should be separated from the DNS API implementation class.

This is necessary for implementation consistency as well as to be resilient to future API changes.